src/main/java/games/strategy/triplea/delegate/UnitComparator.java

```diff
@@ -208,46 +110,14 @@
    */
   public static Comparator<Unit> getUnloadableUnitsComparator(final List<Unit> units, final Route route,
       final PlayerID player) {
-    // compare transports
-    final Comparator<Unit> unloadableTransportsComparator =
-        getUnloadableTransportsComparator(units, route, player, false);
-    // if noTies is set, sort by hashcode so that result is deterministic
-    final Comparator<Unit> movableUnitsComparator = getMovableUnitsComparator(units, route);
-    return (u1, u2) -> {
-      final Unit t1 = TripleAUnit.get(u1).getTransportedBy();
-      final Unit t2 = TripleAUnit.get(u2).getTransportedBy();
-      // check if unloadable units are in a transport
-      if (t1 != null && t2 == null) {
-        return -1;
-      }
-      if (t1 == null && t2 != null) {
-        return 1;
-      }
-      if (t1 != null && t2 != null) {
-        final int compareTransports = unloadableTransportsComparator.compare(t1, t2);
-        if (compareTransports != 0) {
-          return compareTransports;
-        }
-      }
-      // we are sorting air units, or no difference found yet
-      // if noTies is set, sort by hashcode so that result is deterministic
-      return movableUnitsComparator.compare(u1, u2);
-    };
+    return Comparator.comparing(TripleAUnit::get,
+        Comparator.nullsLast(getUnloadableTransportsComparator(units, route, player, false)))
+        .thenComparing(getMovableUnitsComparator(units, route));
```

Comment:
I think this change is causing a bug. Can someone help double check to see that this refactoring is equivalent?

Reply:
@ron-murhammer I spotted the problem.
Will fix shortly

Reply:
Nevermind @ssoloff Already found it😅